spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClientConfiguration.java

```diff
@@ -46,8 +48,8 @@
 	@Bean
 	@ConditionalOnMissingBean
 	public ConsulLifecycle consulLifecycle(ConsulDiscoveryProperties discoveryProperties,
-			HeartbeatProperties heartbeatProperties) {
-		return new ConsulLifecycle(consulClient, discoveryProperties, heartbeatProperties);
+			ConsulConfigProperties configProperties, HeartbeatProperties heartbeatProperties) {
```

Comment:
I'm sorry, I'm a little confused. Are you saying that I need to remove `ConsulConfigProperties` and use a `failFast` boolean flag instead?